apps/fishing-map/features/map/annotations/Annotations.tsx

```diff
@@ -1,93 +1,68 @@
-import { Popup } from 'react-map-gl'
-import { useTranslation } from 'react-i18next'
-import { useSelector } from 'react-redux'
-import {
-  Button,
-  ColorBar,
-  IconButton,
-  InputText,
-  LineColorBarOptions,
-} from '@globalfishingwatch/ui-components'
-import { useEventKeyListener } from '@globalfishingwatch/react-hooks'
+import { HtmlOverlay, HtmlOverlayItem } from '@nebula.gl/overlays'
+import { DragEvent, useCallback } from 'react'
 import { useMapAnnotation, useMapAnnotations } from 'features/map/annotations/annotations.hooks'
-import { DEFAUL_ANNOTATION_COLOR } from 'features/map/map.config'
-import { useLocationConnect } from 'routes/routes.hook'
-import { selectIsGFWUser } from 'features/user/selectors/user.selectors'
-import styles from './Annotations.module.css'
+import { useDeckMap } from '../map-context.hooks'
+import { MapAnnotation, MapAnnotationComponentProps } from './annotations.types'
 
-const colors = [{ id: 'white', value: DEFAUL_ANNOTATION_COLOR }, ...LineColorBarOptions]
-
-const MapAnnotations = () => {
-  const { t } = useTranslation()
-  const { dispatchQueryParams } = useLocationConnect()
-  const gfwUser = useSelector(selectIsGFWUser)
-  const { mapAnnotation, resetMapAnnotation, setMapAnnotation } = useMapAnnotation()
-  const { deleteMapAnnotation, upsertMapAnnotations } = useMapAnnotations()
-  const onConfirmClick = () => {
-    if (!mapAnnotation) {
-      return
-    }
-    upsertMapAnnotations({
-      ...mapAnnotation,
-      id: mapAnnotation.id || Date.now(),
-    })
-    resetMapAnnotation()
-    dispatchQueryParams({ mapAnnotationsVisible: true })
-  }
-  const ref = useEventKeyListener(['Enter'], onConfirmClick)
-
-  if (!mapAnnotation) {
-    return null
-  }
-
-  const onDeleteClick = () => {
-    deleteMapAnnotation(mapAnnotation.id)
-    resetMapAnnotation()
-  }
-
-  if (!gfwUser) {
-    return null
-  }
+const MapAnnotations = (props: MapAnnotationComponentProps): React.ReactNode | null => {
+  const { viewport } = props
+  const { upsertMapAnnotations, mapAnnotations, areMapAnnotationsVisible } = useMapAnnotations()
+  const { setMapAnnotation } = useMapAnnotation()
+  const deck = useDeckMap()
+  const handleHover = useCallback(() => {
+    deck?.setProps({ getCursor: () => 'move' })
+  }, [deck])
+  const handleMouseLeave = useCallback(() => {
+    deck?.setProps({ getCursor: () => 'grab' })
+  }, [deck])
+  const handleDragStart = useCallback(() => {
```

Comment:
What about hiding the annotation while it is being dragged to avoid this weird effect?

https://github.com/GlobalFishingWatch/frontend/assets/10500650/3d2d6cb8-0b02-4e36-bd1c-bbd3e39ccd44